### DISCUSSION TITLE
CMAKE AWS-C-CALC cannot find openssl/evp.h

### DISCUSSION THREAD

**Discussion by BrunoFigRaymarine** [aws/aws-sdk-cpp#3211] [Q&A] [answered]
Any hints as to why my cmake build for android keeps failing with cannot find openssl header files?
I have added the CMAKE_PREFIX_PATH and tried many suggestions on this forum but nothing seems to work :(

**Comment by jmklix** [comment]
What build commands are you trying? 
What version of this sdk are you using?
Did you clone the submodules along with this sdk?
     `git clone --recurse-submodules https://github.com/aws/aws-sdk-cpp`
Did you follow our [build for android documentation](https://github.com/aws/aws-sdk-cpp?tab=readme-ov-file#building-for-android)?

**Reply by BrunoFigRaymarine** [reply, marked as answer]
Hi, After exhaustive trials the only thing that worked was to reverse engineer the MakeFiles and CMakeFiles and set 

-DOPENSSL_INCLUDE_FLAGS
-DOPENSSL_LINKER_FLAGS
-DCMAKE_CXX_FLAGS
-DCMAKE_C_FLAGS

The CMAKE_PREFIX and CMAKE_INCLUDE path were not being considered if you don't build OPENSSL and then you cannot include OPENSSL files when building other submodules
			-DCMAKE_SHARED_LINKER_FLAGS="-L$ANDROID_LIB_PATH -fPIE -pie" \
